### DISCUSSION TITLE
How do I add title sections within the sidebar?

### DISCUSSION THREAD

**Discussion by StephenTangCook** [shuding/nextra#2704] [Q&A] [answered]
For example, the Nextra docs site has "Themes" and "More" sub-sections:

![Screenshot 2024-02-12 at 9 21 12 AM](https://github.com/shuding/nextra/assets/44206254/ec234a1c-87ef-4ecb-852b-4686bdf79fa7)

**Comment by mutasim77** [comment, marked as answer]
If we look at the source code in the `pages/docs/_meta.json` directory, we can see something like this:
```json
{
  "index": "Introduction",
  "guide": "",
  "-- Themes": {
    "type": "separator",
    "title": "Themes"
....
},
```

You may wonder how this works. Look, we are passing a type of `separator`. Now, in your `theme.config.ts` file, we can add something like this to ur configurations:
```ts
const config: DocsThemeConfig = {
  ...,
  sidebar: {
    titleComponent({ title, type }) {
      if (type === 'separator') {
        return <span className="cursor-default">{title}</span>;
      }
      return <>{title}</>;
    },
    defaultMenuCollapseLevel: 1,
    toggleButton: true
  },
}
```
